### DISCUSSION TITLE
On the issue of missing Requirement Bundle

### DISCUSSION THREAD

**Discussion by dongchen1221** [eclipse-tycho/tycho#2364] [Q&A]
[ERROR] Failed to execute goal org.eclipse.tycho:tycho-compiler-plugin:2.7.5:validate-classpath (default-validate-classpath) on project com.nucleisys.ide: Execution default-validate-classpath of goal org.eclipse.tycho:tycho-compiler-plugin:2.7.5:validate-classpath failed: org.osgi.framework.BundleException: Bundle com.nucleisys.ide cannot be resolved:com.nucleisys.ide [42]
[ERROR]   Unresolved requirement: Require-Bundle: org.eclipse.ui
[ERROR] -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.eclipse.tycho:tycho-compiler-plugin:2.7.5:validate-classpath (default-validate-classpath) on project com.nucleisys.ide: Execution default-validate-classpath of goal org.eclipse.tycho:tycho-compiler-plugin:2.7.5:validate-classpath failed: org.osgi.framework.BundleException: Bundle com.nucleisys.ide cannot be resolved:com.nucleisys.ide [42]
  Unresolved requirement: Require-Bundle: org.eclipse.ui

	<repositories>	
		<repository>
		  <id>license-feature</id>
		  <url>https://download.eclipse.org/tools/cdt/releases/9.10</url>
		  <layout>p2</layout>
		</repository>
		<repository>
		  <id>license-feature1</id>
		  <url>https://download.eclipse.org/embed-cdt/releases/5.1.1/p2/</url>
		  <layout>p2</layout>
		</repository>
	</repositories>	

Why is this situation happening? If it's a problem with the target platform, but you can find org.eclipse.ui in my target platform status view？

**Comment by laeubi** [comment]
> but you can find org.eclipse.ui in my target platform status view

Your configuration does not use a target file at all.

So you probably want two things:

1. Use a more recent Tycho version (3.x is the current most recent one)
2. [configure Tycho](https://tycho.eclipseprojects.io/doc/latest/target-platform-configuration/target-platform-configuration-mojo.html) with your target platform
